### PR TITLE
Bugfix: Undefined method 'include?' for nil:NilClass in proxy policy

### PR DESCRIPTION
**Pull request by joroca**

### CODE REVIEW

File: providers/system_proxy.rb [hunk 1]
```diff
@@ -91,10 +92,14 @@
                 fe = Chef::Util::FileEdit.new("/etc/environment")
                 fe.search_file_replace_line(/HTTP_PROXY/i,"HTTP_PROXY=\"#{global_settings['http_proxy']}:#{global_settings['http_proxy_port']}\"")
                 fe.search_file_replace_line(/HTTPS_PROXY/i,"HTTPS_PROXY=\"#{global_settings['https_proxy']}:#{global_settings['https_proxy_port']}\"")
+                fe.write_file
```
Comment (alfem) [inline providers/system_proxy.rb]: is it possible to write this file only once?

File: providers/system_proxy.rb [hunk 2]
```diff
@@ -32,16 +32,17 @@
...
         end
 
-        global_settings['https_proxy'] = if not global_settings['https_proxy'].include?('https://')
-          then global_settings['https_proxy'] = 'https://' + global_settings['https_proxy']
-          else global_settings['https_proxy']
+        if !(global_settings['https_proxy'].nil? || global_settings['https_proxy'].include?('https://'))
+           global_settings['https_proxy'] = 'https://'.concat(global_settings['https_proxy'])
         end
-
+        global_settings['http_proxy'] ||= ''
```
Comment (juanconde) [inline providers/system_proxy.rb]: Default values for global_settings should be assigned before line 35, avoiding the need for testing for nil?. After "http?://" insertion, global_settings should be tested for valid (well formed) URLs and ports. This is a dangerous policy, as it can break connectivity between workstation a Control Centre, and some minimal checks are required.

File: providers/system_proxy.rb (hunk 1 above)
Comment (joroca) [inline providers/system_proxy.rb]: Chef (write_file method) only save the last file_replace_line. It's not a documented behaviour but you can find the same problem on forums, for example:

https://discourse.chef.io/t/inserting-multiple-lines-through-chef-util-fileedit/3676

Comment (alfem) [inline providers/system_proxy.rb]: I am afraid is a bug in Chef 11 code. This method has changed in Chef 12 and (probably) will work as expected in a next GECOS version.

I will accept this PR as is and mark it for a future review.
